### PR TITLE
#2477. Add match operators constants evaluation tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: Language/Expressions/Constants/math_operators_A02_t02.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+extension type const ExtNum(num _) implements num {}
+
+extension type const ExtInt(int _) implements int {}
+
+extension type const ExtDouble(double _) implements double {}
+
+main() {
+  const c1 = ExtNum(1);
+  Expect.identical(-1, -c1);
```
Comment (eernstg) [inline Language/Expressions/Constants/math_operators_A02_t02.dart:43]: But here, `-c1` is not in a constant context so we won't get any errors even if it isn't considered constant. Maybe use `const c1 = -ExtNum(1);`?

Comment (eernstg) [reply Language/Expressions/Constants/math_operators_A02_t02.dart:43]: Similar comment on the other cases.

**Review by eernstg — COMMENTED** [review]
Looks good! Just one thing: I think Language/Expressions/Constants/math_operators_A02_t02.dart needs to be adjusted in order to test a unary minus expression as a constant expression.

**Comment by sgrekhov** [comment]
Indeed. Thank you! Updated

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**